### PR TITLE
HHH-16515 - Add o.h.stat to nullness checking

### PR DESCRIPTION
**Pull request by jrenaat**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java [hunk 1]
```diff
@@ -540,18 +551,29 @@ public String[] getSecondLevelCacheRegionNames() {
...
 	}
 
 	@Override
-	public CacheRegionStatisticsImpl getQueryRegionStatistics(final String regionName) {
-		return l2CacheStatsMap.getOrCompute( regionName, this::computeQueryRegionStatistics );
+	public @Nullable CacheRegionStatisticsImpl getQueryRegionStatistics(final String regionName) {
+		return l2CacheStatsMap.getOrCompute(
+				regionName,
+
+				new Function<>() {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java:567]: Is this really necessary? Other method references seem to work fine as well.

**Review by beikov — APPROVED** [review]

File: hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java (hunk 1 above)
Comment (jrenaat) [reply hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java:567]: Without it, checkerframework complains, it needs the return type of those particular functions to be @Nullable, and that's the only way I could achieve that, which doesn't mean there aren't better ways, I just don't know them ;)

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java:567]: Alright, if that is what it takes, fine then.

File: hibernate-core/src/main/java/org/hibernate/stat/internal/StatsNamedContainer.java [hunk 2]
```diff
@@ -23,7 +26,7 @@
  *
  * @author Sanne Grinovero
  */
-public final class StatsNamedContainer<V> {
+public final class StatsNamedContainer<@Nullable V> {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/stat/internal/StatsNamedContainer.java:29]: This shouldn't be necessary.

File: hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java [hunk 3]
```diff
@@ -101,19 +105,19 @@ public class StatisticsImpl implements StatisticsImplementor, Service {
...
-	private final StatsNamedContainer<CollectionStatisticsImpl> collectionStatsMap = new StatsNamedContainer<>();
+	private final StatsNamedContainer<@Nullable EntityStatisticsImpl> entityStatsMap = new StatsNamedContainer<>();
+	private final StatsNamedContainer<@Nullable NaturalIdStatisticsImpl> naturalIdQueryStatsMap = new StatsNamedContainer<>();
+	private final StatsNamedContainer<@Nullable CollectionStatisticsImpl> collectionStatsMap = new StatsNamedContainer<>();
 
 	/**
 	 * Keyed by query string
 	 */
-	private final StatsNamedContainer<QueryStatisticsImpl> queryStatsMap;
+	private final StatsNamedContainer<@Nullable QueryStatisticsImpl> queryStatsMap;
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java:115]: This shouldn't be necessary.

File: hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java [hunk 4]
```diff
@@ -101,19 +105,19 @@ public class StatisticsImpl implements StatisticsImplementor, Service {
 
 	private final LongAdder optimisticFailureCount = new LongAdder();
 
-	private final StatsNamedContainer<EntityStatisticsImpl> entityStatsMap = new StatsNamedContainer<>();
-	private final StatsNamedContainer<NaturalIdStatisticsImpl> naturalIdQueryStatsMap = new StatsNamedContainer<>();
-	private final StatsNamedContainer<CollectionStatisticsImpl> collectionStatsMap = new StatsNamedContainer<>();
+	private final StatsNamedContainer<@Nullable EntityStatisticsImpl> entityStatsMap = new StatsNamedContainer<>();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java:108]: This shouldn't be necessary.

File: hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java [hunk 5]
```diff
@@ -101,19 +105,19 @@ public class StatisticsImpl implements StatisticsImplementor, Service {
...
+	private final StatsNamedContainer<@Nullable NaturalIdStatisticsImpl> naturalIdQueryStatsMap = new StatsNamedContainer<>();
+	private final StatsNamedContainer<@Nullable CollectionStatisticsImpl> collectionStatsMap = new StatsNamedContainer<>();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java:110]: This shouldn't be necessary.

File: hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java [hunk 6]
```diff
@@ -101,19 +105,19 @@ public class StatisticsImpl implements StatisticsImplementor, Service {
 
 	private final LongAdder optimisticFailureCount = new LongAdder();
 
-	private final StatsNamedContainer<EntityStatisticsImpl> entityStatsMap = new StatsNamedContainer<>();
-	private final StatsNamedContainer<NaturalIdStatisticsImpl> naturalIdQueryStatsMap = new StatsNamedContainer<>();
-	private final StatsNamedContainer<CollectionStatisticsImpl> collectionStatsMap = new StatsNamedContainer<>();
+	private final StatsNamedContainer<@Nullable EntityStatisticsImpl> entityStatsMap = new StatsNamedContainer<>();
+	private final StatsNamedContainer<@Nullable NaturalIdStatisticsImpl> naturalIdQueryStatsMap = new StatsNamedContainer<>();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/stat/internal/StatisticsImpl.java:109]: This shouldn't be necessary.

**Review by beikov — APPROVED** [review]

**Pull request merged**